src/main/scala/org/broadinstitute/dsde/workbench/sam/service/TosService.scala

```diff
@@ -0,0 +1,31 @@
+package org.broadinstitute.dsde.workbench.sam.service
+
+import cats.effect.IO
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.workbench.model.{WorkbenchEmail, WorkbenchGroupName}
+import org.broadinstitute.dsde.workbench.sam.dataAccess.DirectoryDAO
+import org.broadinstitute.dsde.workbench.sam.model.BasicWorkbenchGroup
+import org.broadinstitute.dsde.workbench.sam.util.SamRequestContext
+
+class TosService (val directoryDao: DirectoryDAO) extends LazyLogging {
+
+  def createNewGroupIfNeeded(currentVersion:Int, isEnabled:Boolean): IO[Any] = {
```

Comment:
I think this might be a bit more idiomatic:

```
  def createNewGroupIfNeeded(currentVersion: Int, isEnabled: Boolean): IO[Option[BasicWorkbenchGroup]] = {
    if(isEnabled) {
      getTosGroup(currentVersion).flatMap {
        case Some(_) => IO.none
        case None =>
          logger.info("creating new ToS group")
          directoryDao.createGroup(BasicWorkbenchGroup(WorkbenchGroupName(getGroupName(currentVersion)),
            Set.empty, WorkbenchEmail("foo@bar.com")), samRequestContext = SamRequestContext(None)).map(Option(_))
      }
    } else IO.none
  }
```

And then a second change, turning `groupExists` into `getTosGroup`:

```
  def getTosGroup(currentVersion: Int): IO[Option[BasicWorkbenchGroup]] = {
    directoryDao.loadGroup(WorkbenchGroupName(getGroupName(currentVersion)), SamRequestContext(None))
  }
```

IMO the main benefits are:
1) Keeps everything within an IO context, so there's no need to do any `unsafeRunSync` weirdness which breaks up the execution chain
2) An `Any` return type is a bit sketchy in Scala. This change gives it a concrete return type. While the value isn't actually used upstream, it's nice to have a definitive return type, at least for the sake of readability (IMO)
3) Utilizes pattern matching instead of `isDefined`, which is pretty powerful in Scala and more idiomatic

Reply:
Thanks, Matt, I've incorporated this very useful feedback.